src/AspNetCoreModuleV2/RequestHandlerLib/PipeOutputManager.cpp

```diff
@@ -32,6 +32,31 @@ PipeOutputManager::StopOutputRedirection()
     fflush(stdout);
     fflush(stderr);
 
+    // Restore the original stdout and stderr handles of the process,
+    // as the application has either finished startup or has exited.
+
+    // If stdout/stderr were not set, we need to set it to NUL:
+    // such that other calls to Console.WriteLine don't use an invalid handle
+    FILE *stream;
+
+    if (m_fdPreviousStdOut >= 0)
```

Comment:
could m_fdPreviousStdOut ever be 0?

Reply:
Yeah, I believe stdin is by default set to 0. It's better to be defensive with this check.